File: Design Patterns in Java/Chain of Responsibility Pattern/src/dev/chamalwr/handler/DirectoryHandler.java
package dev.chamalwr.handler;

import dev.chamalwr.model.Email;

import java.util.HashMap;
import java.util.Map;

/**This handler responsible of checking whether the
 * email is available within the directory
**/
public class DirectoryHandler extends EmailHandler{

    Map<String, String> emailDirectory = new HashMap<>();

    public DirectoryHandler() {
        emailDirectory.put("ceo@luxWatches.com", "CEO Office");
        emailDirectory.put("calvin@luxWatches.com", "Chief Designer");
        emailDirectory.put("gosling@luxWatches.com", "Supply Chain Management");
        emailDirectory.put("klein@luxWatches.com", " Engineer");
    }

    @Override
    public void checkMail(Email email) {
        if (emailDirectory.get(email.getReceiver()) != null) {
            System.out.println("Receiver Identified");
            emailHandlerSuccessor.checkMail(email);
        }else{
            System.err.println("Receiver not found!");
        }
    }
}

File: Design Patterns in JavaScript/Chain of Responsibility Pattern/handler/DirectoryHandler.js
const EmailHandler = require("./EmailHandler");

module.exports = class DirectoryHandler extends EmailHandler{
    data = {
         "ceo@luxWatches.com": "CEO Office",
         "calvin@luxWatches.com": "Chief Designer Office",
         "gosling@luxWatches.com": "Supply Chain Management",
         "klein@luxWatches.com": "Engineer Office"
    }

    checkMail(email){
       if (this.data[email.receiver]){
           console.log("Receiver Identified")
           console.log("This will directed to the : " + this.data[email.receiver])
           if (this.successorHandler != null){
               this.successorHandler.checkMail(email)
           }
       }else{
           console.log("Receiver not found!")
       }
    }
}
